torchx/tracker/backend/fsspec.py

```diff
@@ -236,19 +236,33 @@ def __repr__(self) -> str:
         return f"<FsspecTracker: root_path={self._path_builder.path()}>"
 
 
-def _read_config(config_file: str) -> Mapping[str, str]:
+def _put_config(key: str, value: str, config: Dict[str, Any]) -> None:
```

Comment:
This is a form of reduction, if you wish we can convert it into iterative approach something like this:

~~~
keys = key.split(".")

last_key = keys[-1]
parent_keys = keys[:-1]

config = {}
nested_config = config
for parent_key in parent_keys:
  if not parent_key: # takes care of any extra periods
    raise ValueError(f"Illegal config key `{key}`.")
  nested_config.setdefault(parent_key, {})
  nested_config = nested_config[parent_key]
nested_config[last_key] = value
~~~

Reply:
is there a particular reason you prefer the iterative approach to the recursive one?